test/basic_radix2_domain.cpp

```diff
@@ -0,0 +1,111 @@
+//---------------------------------------------------------------------------//
+// Copyright (c) 2024 Dmitrii Tabalin <d.tabalin@nil.foundation>
+//
+// MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+//---------------------------------------------------------------------------//
+
+#include "nil/crypto3/math/algorithms/unity_root.hpp"
+#include "nil/crypto3/math/domains/detail/basic_radix2_domain_aux.hpp"
+#define BOOST_TEST_MODULE basic_radix2_domain_test
+
+#include <vector>
+#include <cstdint>
+
+#include <boost/test/unit_test.hpp>
+#include <boost/test/data/test_case.hpp>
+#include <boost/test/data/monomorphic.hpp>
+
+
+#include <nil/crypto3/algebra/fields/arithmetic_params/bls12.hpp>
+#include <nil/crypto3/math/algorithms/make_evaluation_domain.hpp>
+#include <nil/crypto3/math/polynomial/polynomial.hpp>
+#include <nil/crypto3/math/polynomial/polynomial_dfs.hpp>
+#include <nil/crypto3/math/polynomial/shift.hpp>
+
+#include <nil/crypto3/algebra/random_element.hpp>
+
+using namespace nil::crypto3::algebra;
+using namespace nil::crypto3::math;
+
+typedef fields::bls12_fr<381> FieldType;
+
+BOOST_AUTO_TEST_SUITE(basic_radix2_domain_test_suit)
+
+BOOST_AUTO_TEST_CASE(asic_radix2_domain_benchmark) {
```

Comment:
If this test runs too long, like > 30 seconds, add *boost::unit_test::disabled() at the end, so we can manually run it, but it does not take too much time on the CI.

Reply:
I've disabled this (and another one I've added); for me it runs in ~20 seconds, but might be slower on CI machine.